src/ai/backend/storage/weka/weka_client.py

```diff
@@ -0,0 +1,388 @@
+            response = await self._build_request(sess, "GET", "/fileSystems")
+            data = await response.json()
+        if len(data["data"]) == 0:
+            raise WekaNotFoundError
+        return [WekaFs.from_json(fs_json) for fs_json in data["data"]]
+
+    @error_handler
+    async def get_fs(self, fs_uid: str) -> WekaFs:
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(sess, "GET", f"/fileSystems/{fs_uid}")
+            data = await response.json()
+        if data.get("data") is None:
+            raise WekaNotFoundError
+        return WekaFs.from_json(data["data"])
+
+    @error_handler
+    async def list_quotas(self, fs_uid: str) -> Iterable[WekaQuota]:
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(
+                sess,
+                "GET",
+                f"/fileSystems/{fs_uid}/quota",
+            )
+            data = await response.json()
+        if isinstance(data['data'], list):
+            return [
+                WekaQuota.from_json(quota_info["quota_id"], quota_info)
+                for quota_info in data["data"]
+            ]
+        else:
+            return [
+                WekaQuota.from_json(quota_id, data["data"][quota_id])
+                for quota_id in data["data"].keys()
+            ]
+
+    @error_handler
+    async def get_quota(self, fs_uid: str, inode_id: int) -> WekaQuota:
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(
+                sess,
+                "GET",
+                f"/fileSystems/{fs_uid}/quota/{inode_id}",
+            )
+            data = await response.json()
+        if len(data["data"].keys()) == 0:
+            raise WekaNotFoundError
+        quota_id = data["data"].keys()[0]
+        return WekaQuota.from_json(quota_id, data["data"][quota_id])
+
+    @error_handler
+    async def set_quota(
+        self,
+        fs_uid: str,
+        inode_id: int,
+        soft_limit: BinarySize,
+        hard_limit: BinarySize,
+    ) -> WekaQuota:
+        def _format_size(s: BinarySize) -> str:
+            ss = str(s)
+            if ss.endswith("bytes"):
+                return ss.replace("bytes", "B").replace(" ", "")
+            else:
+                return ss.replace(" ", "")
+
+        body = {
+            "grace_seconds": None,
+            "soft_limit_bytes": _format_size(soft_limit),
+            "hard_limit_bytes": _format_size(hard_limit),
+        }
+
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(
+                sess,
+                "PUT",
+                f"/fileSystems/{fs_uid}/quota/{inode_id}",
+                body,
```

Comment:
In [L287](https://github.com/lablup/backend.ai/pull/443/files#diff-36cfe4b50d0a1c978df8cb77d6e45162c6df438ed66997fc94e8d231218b00cbR287), sending Mapping type value triggers parsing error. Let's wrap with `json.dumps()` function so that it will automatically be stringified.

Reply:
I think we can solve this easily, by changing `body` keyword parameter of `func()` call in L193 to `json`. Check https://docs.aiohttp.org/en/stable/client_advanced.html#custom-request-headers.